evm_arithmetization/src/witness/transition.rs

```diff
@@ -383,44 +462,55 @@ pub(crate) fn fill_stack_fields<F: Field>(
                 return Err(ProgramError::InterpreterError);
             }
         }
-        state.registers.check_overflow = false;
     }
+    state.registers.check_overflow = false;
 
     Ok(())
 }
 
-fn try_perform_instruction<F: Field>(
-    state: &mut GenerationState<F>,
-) -> Result<Operation, ProgramError> {
+fn try_perform_instruction<F: Field>(any_state: &mut State<F>) -> Result<Operation, ProgramError> {
+    let is_generation = any_state.is_generation_state();
+    let registers = any_state.get_registers();
+
+    let state = any_state.get_mut_generation_state();
     let (mut row, opcode) = base_row(state);
```

Comment:
I think here you will push operations even for the interpreter, because `base_row` calls `mem_read_code_with_log_and_fill`. It's also odd that the interpreter must also need to generate a row.